chain-impl-mockchain/src/block/version.rs

```diff
@@ -68,10 +70,25 @@ impl BlockVersion {
     }
 }
 
-#[derive(Debug, Clone, Copy, FromPrimitive, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    Display,
+    EnumString,
+    FromPrimitive,
+    IntoStaticStr,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+)]
+#[strum(serialize_all = "snake_case")]
 pub enum ConsensusVersion {
     None = 0,
     Bft = 1,
+    #[strum(serialize = "genesis")]
     GenesisPraos = 2,
```

Comment:
I am not sure `Strum` was needed just for this. Worse, having an automatically generated value for what is in the block does not inspire me confidence. I'd rather like to have the code written hand here (especially it is rather simple here)

Reply:
We need both parsing and printing textual representation. `Strum` lets the implementation be DRY, it's impossible to make mistake now and we have less boilerplate. It's exactly like `num_derive`, which we already use, but for strings.

You're right about generated values being dangerously cryptic. How about explicitly defining every variant representation like in `GenesisPraos`?